app/master/cluster_master.py

```diff
@@ -62,12 +63,15 @@ def api_representation(self):
             'slaves': slaves_representation,
         }
 
-    def builds(self):
+    def builds(self, start=None, amount=None):
         """
         Returns a list of all builds
         :rtype: list[Build]
         """
-        return self._all_builds_by_id.values()
+        num_builds = len(self._all_builds_by_id)
+        start = start or 0
+        amount = amount or num_builds
+        return [self._all_builds_by_id[key] for key in islice(self._all_builds_by_id, start, min((start + amount), num_builds))]
```

Comment:
This is slightly difficult to read, I think this should separated out into a few lines, maybe something like this (untested):
```python
total_builds = len(self._all_builds_by_id)
page = page or 0
size = size or total_builds
stop_index = min((page + size), total_builds))
builds_on_request_page = islice(self._all_builds_by_id, start=page, stop=stop_index)
return builds_on_request_page.values()
```